src/Compilers/Core/Portable/InternalUtilities/ConsList`1.cs

```diff
@@ -34,7 +39,9 @@ public T Current
                 get
                 {
                     Debug.Assert(_tail != null);
-                    return _current;
+
+                    // This never returns null after a proper call to `MoveNext` returned true.
```

Comment:
Consider adding an assert instead of a suppression.

Reply:
➡️ Fixed